core/header.cpp

```diff
@@ -169,6 +169,15 @@ namespace MR
       if (App::project_version)
         H.keyval()["project_version"] = App::project_version;
 
+      std::string cmd = App::argv[0];
+      for (int n = 1; n < App::argc; ++n)
+        cmd += std::string(" \"") + App::argv[n] + "\"";
```

Comment:
Yes, it is a bit ugly. But white spaces aren't the only potential problem. We'd have to watch for at least these characters: ` []*?%#~&$(){}\t\\"';:<>` - i.e. absolutely _anything_ that the shell might conceivably interpret. If the command we received contained these characters, then that's what the user intended, and they must have supplied the argument in such as way as to avoid interpretation by the shell. So we should respect that - the arguments the executable received have already been interpreted by the shell, so we should ensure that a straight copy/paste of the command works without the shell re-interpreting anything incorrectly. 

Also, I think we should be using single quotes `'` to avoid further variable expansion, etc. - otherwise an argument like `stuff$n.mif` would be interpreted when it should be left alone (although I completely agree that users who actually names their files like that deserve everything they get...).